Source/WebCore/page/LocalFrameView.cpp

```diff
@@ -6278,6 +6282,15 @@ OverscrollBehavior LocalFrameView::verticalOverscrollBehavior()  const
     return OverscrollBehavior::Auto;
 }
 
+ScrollbarWidth LocalFrameView::scrollbarWidth()  const
+{
+    auto* document = m_frame->document();
+    auto scrollingObject = document && document->documentElement() ? document->documentElement()->renderer() : nullptr;
```

Comment:
Is `documentElement` the right thing to check for the style? How about `scrollingElement`? Does `scrollbar-width` propagate from the body to the document like `overflow` does?

Reply:
This reads to me like the answer is no it doesn't propagate and as such I think you're right it should be scrollingElement not document element.

![image](https://github.com/WebKit/WebKit/assets/32498324/d330e4a5-eeee-4a28-a0a2-3893815d404e)

Reply:
Make sure there at WPT tests that test that.

Reply:
@lukewarlow ^ can you please do this as a followup?